### PR TITLE
dev-db/sadisplay: version bump 0.4.9, bug #718272

### PR DESCRIPTION
**Pull request by TheChymera**
and PYTHON_COMPAT update

Package-Manager: Portage-2.3.99, Repoman-2.3.22
Signed-off-by: Horea Christian <chr@chymera.eu>

### CODE REVIEW

File: dev-db/sadisplay/sadisplay-0.4.9.ebuild [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+RDEPEND="dev-python/sqlalchemy[${PYTHON_USEDEP}]"
+DEPEND="
+	dev-python/setuptools[${PYTHON_USEDEP}]
+	test? ( ${RDEPEND} )
+	"
+#S="${WORKDIR}/estin-${PN}-${COMMIT_ID}"
+
+python_test() {
+	nosetests -v || die
+}
```
Comment (juippis) [inline dev-db/sadisplay/sadisplay-0.4.9.ebuild:29]: Looks like this should be doable using *distutils-r1.eclass*.

File: dev-db/sadisplay/sadisplay-0.4.9.ebuild [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+KEYWORDS="~amd64 ~x86"
+IUSE="test"
+RESTRICT="!test? ( test )"
+
+RDEPEND="dev-python/sqlalchemy[${PYTHON_USEDEP}]"
+DEPEND="
+	dev-python/setuptools[${PYTHON_USEDEP}]
+	test? ( ${RDEPEND} )
+	"
+#S="${WORKDIR}/estin-${PN}-${COMMIT_ID}"
```
Comment (juippis) [inline dev-db/sadisplay/sadisplay-0.4.9.ebuild:25]: This comment can be removed.

File: dev-db/sadisplay/sadisplay-0.4.9.ebuild [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="test"
+RESTRICT="!test? ( test )"
+
+RDEPEND="dev-python/sqlalchemy[${PYTHON_USEDEP}]"
+DEPEND="
+	dev-python/setuptools[${PYTHON_USEDEP}]
+	test? ( ${RDEPEND} )
```
Comment (juippis) [inline dev-db/sadisplay/sadisplay-0.4.9.ebuild:23]: Both handled via *distutils-r1.eclass* nowadays.

**Comment by TheChymera** [comment]
@juippis ok, now?

File: dev-db/sadisplay/sadisplay-0.4.9.ebuild [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+KEYWORDS="~amd64 ~x86"
+IUSE="test"
+RESTRICT="!test? ( test )"
+
+RDEPEND="dev-python/sqlalchemy[${PYTHON_USEDEP}]"
+DEPEND="
+	dev-python/setuptools[${PYTHON_USEDEP}]
+	"
+
+distutils_enable_tests nose
```
Comment (juippis) [inline dev-db/sadisplay/sadisplay-0.4.9.ebuild:25]: `Failure: ModuleNotFoundError (No module named 'pytest') ... ERROR`

You can still add test dependencies while using `distutils_enable_tests nose` if it really requires both. I tested. after installing pytest everything works.

File: dev-db/sadisplay/sadisplay-0.4.9.ebuild [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="test"
+RESTRICT="!test? ( test )"
+
+RDEPEND="dev-python/sqlalchemy[${PYTHON_USEDEP}]"
+DEPEND="
+	dev-python/setuptools[${PYTHON_USEDEP}]
+	"
```
Comment (juippis) [inline dev-db/sadisplay/sadisplay-0.4.9.ebuild:23]:
```
 * DISTUTILS_USE_SETUPTOOLS value is probably incorrect
 *   value:    DISTUTILS_USE_SETUPTOOLS=bdepend (default?)
 *   expected: DISTUTILS_USE_SETUPTOOLS=rdepend
```

Use the eclass variable, and you can remove whole `DEPEND` from this ebuild.

File: dev-db/sadisplay/sadisplay-0.4.9.ebuild [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+inherit distutils-r1
+
+DESCRIPTION="SqlAlchemy schema display script"
+HOMEPAGE="https://bitbucket.org/estin/sadisplay/wiki/Home"
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="test"
```
Comment (TheChymera) [inline dev-db/sadisplay/sadisplay-0.4.9.ebuild:19]: @juippis ok now, or should these also go?

**Comment by TheChymera** [comment]
@juippis why, shouldn't it be bdepend?

**Comment by juippis** [comment]
```
 * Messages for package dev-db/sadisplay-0.4.9:

 * DISTUTILS_USE_SETUPTOOLS value is probably incorrect
 *   value:    DISTUTILS_USE_SETUPTOOLS=bdepend (default?)
 *   expected: DISTUTILS_USE_SETUPTOOLS=rdepend
```

Your package most likely uses entry_points to determine its runtime dependencies.

**Pull request closed without merging**